### PR TITLE
#164798035 admin should be able to assign a user as a reviewer

### PR DESCRIPTION
**Pull request by amaechi-chuks**
#### Description
Role-based access control functionality. Admin can assign a user to be a reviewer.

#### Type of change

- [x] New Functionality(Role based access control functionality)

#### How Has This Been Tested?

- [x] Unit testing
- [x] End to end testing

#### Checklist:
- [x] create a function that can allow an admin to assign a user as a reviewer
- [x] write test cases

#### PT-ID

[#164798035 ](https://www.pivotaltracker.com/story/show/164798035)

#### Screenshots
**Activate a user be reviewer**
image

**Deactivate a user from being a reviewer**
image

#### Background:
N/A

#### Questions:
N/A

### CODE REVIEW

File: server/controllers/admin.controllers.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+const deactivateReviewer = async (req, res) => {
+  const adminId = req.user.userObj.id;
+  const userId = req.params.id;
+  try {
+    // Is user an admin?
+    if (adminId === userId) {
+      return res.status(400).json({
+        errors: {
+          body: ['User is not a reviewer'],
+        },
```
Comment (rotimi-babalola) [inline server/controllers/admin.controllers.js]: Update this error message

File: tests/admin.test.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../server/app';
+
+chai.use(chaiHttp);
+let token1;
+describe('TEST LIKE', () => {
+  it('lgoin a user', done => {
```
Comment (rotimi-babalola) [inline tests/admin.test.js]: Should be `login`

File: server/controllers/admin.controllers.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+ */
+/**
+ * @method activateReviewer
+ * @description Admin priviledges
+ * @param {Object} req
+ * @param {Object} res
+ * @returns {undefined}
+ */
+const activateReviewer = async (req, res) => {
+  // const adminId = req.user.userObj.id;
```
Comment (rotimi-babalola) [inline server/controllers/admin.controllers.js]: If this code is commented out, we might as well remove it @amaechi-chuks

File: server/controllers/admin.controllers.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+};
+/**
+ * @method deactivateReviewer
+ * @description Admin priviledges
+ * @param {*} req
+ * @param {*} res
+ * @returns {undefined}
+ */
+const deactivateReviewer = async (req, res) => {
+  // const adminId = req.user.userObj.id;
```
Comment (rotimi-babalola) [inline server/controllers/admin.controllers.js]: Same as ☝️

File: tests/admin.test.js (hunk 2 above)
Comment (rotimi-babalola) [reply tests/admin.test.js]: Also fix this typo @amaechi-chuks

File: server/controllers/admin.controllers.js (hunk 3 above)
Comment (amaechi-chuks) [reply server/controllers/admin.controllers.js]: Noted

File: server/controllers/admin.controllers.js (hunk 4 above)
Comment (amaechi-chuks) [reply server/controllers/admin.controllers.js]: Noted

File: tests/admin.test.js (hunk 2 above)
Comment (amaechi-chuks) [reply tests/admin.test.js]: Noted

**Pull request closed without merging**